governance/program/src/state/realm_config.rs

```diff
@@ -0,0 +1,108 @@
+//! RealmConfig account
+
+use solana_program::{
+    account_info::AccountInfo, program_error::ProgramError, program_pack::IsInitialized,
+    pubkey::Pubkey,
+};
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+
+use crate::{
+    error::GovernanceError,
+    state::enums::GovernanceAccountType,
+    tools::account::{get_account_data, AccountMaxSize},
+};
+
+/// RealmConfig account
+/// The account is an optional extension to RealmConfig stored on Realm account
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub struct RealmConfigAccount {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// The realm the config belong to
+    pub realm: Pubkey,
+
+    /// Addin providing voter weights for community token
+    pub community_voter_weight: Option<Pubkey>,
+
+    /// Reserved for community max vote weight addin
+    pub reserved_1: Option<Pubkey>,
```

Comment:
What is this reserved for?

Reply:
The `add-in` would allow to provide `max voter weight `externally. Currently the `governance` program uses the governing token mint supply as the max voter weight but there are scenarios when this is not sufficient:

- When `voter weight add-in `is used it's common that the max voter weight is also affected. For example the `voter weight add-in` might add scaling factors based on the token lock period, which it turns changes the max possible weight. 
Note: In this scenario the `voter weight addin` and `max voter weight addin` could be both implemented by the same program.
- When the governance token is fully minted but the full supply is not in circulation yet. In this case the `max voter weight addin` would provide the actual circulating supply.

Reply:
It can be the same program but it's not necessary. You can also have just one of the `addins`.
For example the common use case for just `max vote weight addin` is a governance token which is fully minted but not circulating and the locked none distributed supply is stored in a treasury account. In this case you can create a very simple reusable addin which can provide:  `max_vote_weight` = `mint_supply` - `treasury_amount`